docs/operations/scalable-query-frontend.md

```diff
@@ -5,25 +5,52 @@ weight: 5
+which are each capable of executing a query.
+Therefore, scaling the query frontend impacts the amount of work each individual querier is attempting to do at any given time.
 
-## Querier Max Concurrency
+Scaling up may cause a querier to attempt more work than they are configured for due to restrictions such as memory and cpu limits.
+Additionally, the PromQL engine itself is limited in the number of queries it can do as configured by the `-querier.max-concurrent` parameter.
+Attempting more queries concurrently than this value causes the queries to queue up in the querier itself.
 
-Make sure to configure the querier frontend worker to match max concurrency.  This will allow the operator to freely scale the frontend up and down without impacting the amount of work an individual querier is attempting to perform.  More details [here](https://cortexmetrics.io/docs/proposals/scalable-query-frontend/#dynamic-querier-concurrency).
+For similar reasons scaling down the query frontend may cause a querier to not use its allocated memory and cpu effectively.
+This will lower effective resource utilization.
+Also, because individual queriers will be doing less work, this may cause increased queueing in the query frontends.
 
-### Example Configuration
+### Querier Max Concurrency
 
-**CLI**
-```
--querier.worker-match-max-concurrent=true
-```
+To guarantee that querier doesn't receive more queries that it can handle at the same time, make sure to configure the querier to match its PromQL concurrency with number of connections.
+This can be done by using `-querier.worker-match-max-concurrent=true` option, or `match_max_concurrent: true` field in `frontend_worker` section of YAML config file.
+This allows the operator to freely scale the frontend or scheduler up and down without impacting the amount of work an individual querier is attempting to perform.
 
-**Config File**
-```yaml
-frontend_worker:
-  match_max_concurrent: true
-```
\ No newline at end of file
+### Query Scheduler
+
+Query scheduler is a service that moves the in-memory queue from query frontend to a separate component.
+This makes scaling query frontend easier, as it allows running multiple query frontends without increasing the number of queues.
+
+In order to use query scheduler, both query frontend and queriers must be configured with query scheduler address
+(using `-frontend.scheduler-address` and `-querier.scheduler-address` options respectively).
+
+Note that querier will only fetch queries from query frontend or query scheduler, but not both.
+`-querier.frontend-address` and `-querier.scheduler-address` options are mutually exclusive, and at most one can be set.
+
+When using query scheduler, it is recommended to run two query scheduler instances.
+Running only one query scheduler poses a risk of increased query latency when single scheduler crashes or restarts.
```

Comment:
> poses a risk of increased query latency

Increased query latency or failing queries? If the query-frontend can't enqueue a query in the query-scheduler, doesn't the query fail?

Reply:
If frontend has no workers attempting to forward messages to scheduler (eg. because no scheduler is running), or frontend workers are unable to connect (send "INIT" message and get "OK" back), then frontend will get stuck while trying to pass query to the channel -- but no worker is reading from the channel. Once workers connect to schedulers, they will process the query. From user's perspective, query is just slower.

If workers are connected and read from channel, then forwarding of the query to scheduler may fail. In this case frontend will retry to send query to the channel. This adds latency. If this forwarding fails too many times, query fails.